src/libsystemd/sd-event/sd-event.c

```diff
@@ -3379,7 +3383,8 @@ static int event_source_enter_ratelimited(sd_event_source *s) {
 
         event_source_pp_prioq_reshuffle(s);
 
-        log_debug("Event source %p (%s) entered rate limit state.", s, strna(s->description));
+        if (exceeded)
+                log_debug("Event source %p (%s) entered rate limit state.", s, strna(s->description));
```

Comment:
why conditionalize this?

Reply:
That's to distinguish source limiting based on real even rate vs artificial via explicit call.